HW04/Dmitry_Susha/lib/homework.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require_relative 'homework_observable'
+require_relative 'notification'
+class Homework
+  include HomeworkObservable
+  include Notification
+  attr_reader :title, :description, :student
+  attr_accessor :state
+
+  def initialize(title:, description:, student:, state: 'Created')
+    super()
+    @title = title
+    @description = description
+    @student = student
+    @state = state
+    add_notification(student, "You have new homework: #{title}")
+    add_homework(student)
```

Comment:
we usually don't add such methods to initializer, we just call them on the object
```rb
homework = Homework.new(title: title, description: description, student: student)
homework.add_notification
homework.add_homework

```